biogeochem/EDCohortDynamicsMod.F90

```diff
@@ -124,12 +152,39 @@ subroutine create_cohort(currentSite, patchptr, pft, nn, hite, dbh, bleaf, bfine
     new_cohort%canopy_layer = clayer
     new_cohort%canopy_layer_yesterday = real(clayer, r8)
     new_cohort%laimemory    = laimemory
-    new_cohort%bdead        = bdead
-    new_cohort%bstore       = bstore
-    new_cohort%bl           = bleaf
-    new_cohort%br           = bfineroot
-    new_cohort%bsw          = bsap
-    new_cohort%ode_opt_step = 1.0e6_r8   ! Initialize the integrator step size as super-huge
+
+
+    ! Initialize the Plant allocative Reactive Transport (PaRT) module
+    ! Choose from one of the extensible hypotheses (EH)
+    ! -----------------------------------------------------------------------------------
+
+    call InitPRTCohort(new_cohort)
+
+    ! The initialization allocates memory, but the boundary and initial
+    ! contitions must be set. All new cohorts go through create_cohort()
+    ! so this should be the only place this is called.  Alternatively
+    ! cohorts can be copied and fused, but special routines handle that.
+    ! -----------------------------------------------------------------------------------
+
+    select case(hlm_parteh_model)
```

Comment:
I've always found it hard to remember how these 'select case' things work, and it's difficult to discover where to look from first principles. Maybe add a note to the effect of "you can figure out which case you are using by looking at the 'x' settings in 'y' place..." Apologies for my backwardsness...

Reply:
ok, will do